extensions/binding/org.eclipse.smarthome.binding.fsinternetradio/src/main/java/org/eclipse/smarthome/binding/fsinternetradio/internal/radio/FrontierSiliconRadioConnection.java

```diff
@@ -0,0 +1,207 @@
+ */
+package org.eclipse.smarthome.binding.fsinternetradio.internal.radio;
+
+import java.io.IOException;
+
+import org.apache.commons.httpclient.DefaultHttpMethodRetryHandler;
+import org.apache.commons.httpclient.HttpClient;
+import org.apache.commons.httpclient.HttpException;
+import org.apache.commons.httpclient.HttpMethod;
+import org.apache.commons.httpclient.HttpStatus;
+import org.apache.commons.httpclient.methods.GetMethod;
+import org.apache.commons.httpclient.params.HttpMethodParams;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class holds the http-connection and session information for controlling the radio.
+ *
+ * @author Rainer Ostendorf
+ * @author Patrick Koenemann
+ */
+public class FrontierSiliconRadioConnection {
+
+    private static final Logger logger = LoggerFactory.getLogger(FrontierSiliconRadioConnection.class);
+
+    /** Timeout for HTTP requests. */
+    private final static int SOCKET_TIMEOUT = 5000; // ms
+
+    /** Hostname of the radio. */
+    private final String hostname;
+
+    /** Port number, usually 80. */
+    private final int port;
+
+    /** Access pin, passed upon login as GET parameter. */
+    private final String pin;
+
+    /** The session ID we get from the radio after logging in. */
+    private String sessionId;
+
+    /** http clients, store cookies, so it is kept in connection class. */
+    private HttpClient httpClient = null;
+
+    /** Flag indicating if we are successfully logged in. */
+    private boolean isLoggedIn = false;
+
+    public FrontierSiliconRadioConnection(String hostname, int port, String pin) {
+        this.hostname = hostname;
+        this.port = port;
+        this.pin = pin;
+    }
+
+    /**
+     * Perform login/establish a new session. Uses the PIN number and when successful saves the assigned sessionID for
+     * future requests.
+     *
+     * @return <code>true</code> if login was successful; <code>false</code> otherwise.
+     * @throws IOException if communication with the radio failed, e.g. because the device is not reachable.
+     */
+    public boolean doLogin() throws IOException {
+        isLoggedIn = false; // reset login flag
+
+        if (httpClient == null) {
+            httpClient = new HttpClient();
+        }
+
+        final String url = "http://" + hostname + ":" + port + "/fsapi/CREATE_SESSION?pin=" + pin;
+
+        logger.trace("opening URL:" + url);
+
+        final HttpMethod method = new GetMethod(url);
+        method.getParams().setSoTimeout(SOCKET_TIMEOUT);
+        method.getParams().setParameter(HttpMethodParams.RETRY_HANDLER, new DefaultHttpMethodRetryHandler(3, false));
+
+        try {
+            final int statusCode = httpClient.executeMethod(method);
+            if (statusCode != HttpStatus.SC_OK) {
+                logger.warn("Method failed: " + method.getStatusLine());
+                if (method.getStatusCode() == 403) {
```

Comment:
in my case I have a 404 - is it here ok to not throw an exception, but to continue in the code?
Probably not - whenever you do not get a SC_OK, you should probably throw an exception here.